src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -297,318 +300,117 @@
       "Login thread will sleep until the specified window factor of time from last refresh to "
       + "ticket's expiry has "
       + "been reached, at which time it will try to renew the ticket.";
+  public static final String KAFKA_REST_RESOURCE_EXTENSION_DOC =
+      "Fully qualified class name of a  valid Implementation of the interface RestResourceExtension"
+      + "This can be used to inject user defined resources like filters. Typically used to add "
+      + "custom capability like logging, security, etc  ";
   private static final boolean ZOOKEEPER_SET_ACL_DEFAULT = false;
   private static final ConfigDef config;
 
   static {
-    config = baseConfigDef()
-        .defineOverride(
-            PORT_CONFIG,
-            ConfigDef.Type.INT,
-            KAFKAREST_PORT_DEFAULT,
-            ConfigDef.Importance.LOW,
-            PORT_CONFIG_DOC
-        )
-        .defineOverride(
-            LISTENERS_CONFIG,
-            ConfigDef.Type.LIST,
-            KAFKAREST_LISTENERS_DEFAULT,
-            ConfigDef.Importance.HIGH,
-            LISTENERS_DOC
-        )
-        .defineOverride(
-            RESPONSE_MEDIATYPE_PREFERRED_CONFIG,
-            Type.LIST,
-            Versions.PREFERRED_RESPONSE_TYPES,
-            Importance.LOW,
-            RESPONSE_MEDIATYPE_PREFERRED_CONFIG_DOC
-        )
-        .defineOverride(
-            RESPONSE_MEDIATYPE_DEFAULT_CONFIG,
-            Type.STRING,
-            Versions.KAFKA_MOST_SPECIFIC_DEFAULT,
-            Importance.LOW,
-            RESPONSE_MEDIATYPE_DEFAULT_CONFIG_DOC
-        )
-        .defineOverride(
-            METRICS_JMX_PREFIX_CONFIG,
-            Type.STRING,
-            METRICS_JMX_PREFIX_DEFAULT_OVERRIDE,
-            Importance.LOW,
-            METRICS_JMX_PREFIX_DOC
-        )
+    config = baseKafkaRestConfigDef();
+  }
+
+  protected static ConfigDef baseKafkaRestConfigDef() {
```

Comment:
maybe put an entry in checkstyle/suppressions.xml for MethodLength instead of reformatting to something that is harder to read?